tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -79,43 +112,66 @@ def test_vm_snapshot_ops(
-        # Stopping VM before taking snapshot of the VM PVC
-        vm_obj.stop()
-        # Taking Snapshot of PVC
-        pvc_obj = vm_obj.get_vm_pvc_obj()
-        snap_obj = snapshot_factory(pvc_obj)
-        # Restore the snapshot
-        res_snap_obj = snapshot_restore_factory(
-            snapshot_obj=snap_obj,
-            storageclass=vm_obj.sc_name,
-            size=vm_obj.pvc_size,
-            volume_mode=snap_obj.parent_volume_mode,
-            access_mode=vm_obj.pvc_access_mode,
-            status=constants.STATUS_BOUND,
-            timeout=300,
+        vm_objs_def, vm_objs_aggr, _, _ = multi_cnv_workload(
+            namespace=proj_obj.namespace
         )
-        # Create new VM using the restored PVC
-        res_vm_obj = cnv_workload(
-            volume_interface=constants.VM_VOLUME_PVC,
-            source_url=constants.CNV_FEDORA_SOURCE,
-            existing_pvc_obj=res_snap_obj,
-            namespace=vm_obj.namespace,
-        )[1]
-        # Write new file to VM
-        run_dd_io(vm_obj=res_vm_obj, file_path=file_paths[1], verify=True)
-        # Validate data integrity of file written before taking snapshot
-        res_csum = cal_md5sum_vm(vm_obj=res_vm_obj, file_path=file_paths[0])
-        assert (
-            source_csum == res_csum
-        ), f"Failed: MD5 comparison between source {vm_obj.name} and cloned {res_vm_obj.name} VMs"
-        res_vm_obj.stop()
+        vm_list = vm_objs_def + vm_objs_aggr
+
+        for vm_obj in vm_list:
+            # Expand PVC if `pvc_expand_before_snapshot` is True
+            pvc_obj = vm_obj.get_vm_pvc_obj()
+            if pvc_expand_before_snapshot:
+                pvc_obj.resize_pvc(new_size=50, verify=True)
+
+            # Writing IO on source VM
+            source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
+
+            # Stopping VM before taking snapshot of the VM PVC
+            vm_obj.stop()
+
+            # Taking Snapshot of PVC
+            pvc_obj = vm_obj.get_vm_pvc_obj()
+            snap_obj = snapshot_factory(pvc_obj)
+
+            # Restore the snapshot
+            res_snap_obj = snapshot_restore_factory(
+                snapshot_obj=snap_obj,
+                storageclass=vm_obj.sc_name,
+                volume_mode=snap_obj.parent_volume_mode,
+                access_mode=vm_obj.pvc_access_mode,
+                status=constants.STATUS_BOUND,
+                timeout=300,
+            )
+
+            # Create new VM using the restored PVC
+            res_vm_obj = cnv_workload(
+                source_url=constants.CNV_FEDORA_SOURCE,
+                storageclass=vm_obj.sc_name,
+                existing_pvc_obj=res_snap_obj,
+                namespace=vm_obj.namespace,
+            )[-1]
+
+            # Expand PVC if `pvc_expand_after_restore` is True
+            if pvc_expand_after_restore:
+                new_size = 50
+                res_snap_obj.resize_pvc(new_size=new_size, verify=True)
```

Comment:
Verify whether resize of the disk is reflected inside the VM via df -h or other cmds.

Reply:
Added.